### PR TITLE
Add some more tests of NodeList iterability

### PR DESCRIPTION
**Pull request by domenic**
Adapted from https://github.com/tmpvar/jsdom/blob/8b82e0c39bf856b072130e99cc60ca06ed75f7a7/test/living-dom/node-list.js.

### CODE REVIEW

File: dom/nodes/NodeList-Iterable.html [hunk 1]
```diff
@@ -34,4 +36,24 @@
...
+        let total = 0;
+        for (const el of nodeList) {
+            ++total;
+            assert_equals(el.localName, "b");
+            if (total < 3) {
+                container.appendChild(document.createElement("b"));
+            }
+        }
+
+        assert_equals(total, 5);
```
Comment (rniwa) [inline dom/nodes/NodeList-Iterable.html]: Can we also confirm the order in which elements appear?
e.g. we can add class/id/title on each `b` to identify each b's.

Comment (domenic) [inline dom/nodes/NodeList-Iterable.html]: Done

**Comment by domenic** [comment]
No idea what the stability bot's problem is this time...

**Comment by domenic** [comment]
OK, bot problems were my bad messing up submodules. Good for a new review.

**Comment by rniwa** [comment]
r=me.

**Pull request merged**